Update Haskell package set to Stackage Nightly 2021-04-06 (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on **Friday, 2021-04-16 20:00 UTC+2**. Everyone can participate in that process live on Twitch at https://www.twitch.tv/peti343. In addition to the chat features offered by Twitch, there is also a voice conference at https://discord.gg/YTEa3XR viewers can use to chat with me and with each other.

cdepillabout committed:
haskellPackages: add changelog argument to Haskell generic-builder.nix

changelog was recently added as a new meta field in
https://github.com/NixOS/nixpkgs/pull/60371.

This commit adds similar support to the Haskell generic builder.

cdepillabout committed:
spago: make use of new changelog field for haskell generic builder

peti committed:
Stackage Nightly 2021-04-06

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-2-g2ef7e72 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/1aad60ed9679a7597f3fc3515a0fe26fdb896e55.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-2-g2ef7e72 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/3601914c12f130f2ed460b8531502da57e7ef6b1.

guibou committed:
haskellPackages.PyF: un mark as broken

`PyF` is not broken, neither `krank` which depends on it.

peti committed:
Merge origin/master into haskell-updates.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/46b74aa1f540799cf40a1b5481afab2bdac3084d.

sternenseemann committed:
haskellPackages.essence-of-live-coding-warp: fix eval and build

No idea why essence-of-live-coding-warp constrains on >= 0.2.5, but if
it breaks something jailbreaking this then it's on them for not
following PVP, I guess.

maralorn commented:
I just wanted to remark that HLS 1.1.0 is at this stage half-released to hackage. That means we need to pin down all already released dependencies for 1.1.0 so that they stay compatible with HLS 1.0.0. This shouldn‘t be hard to do, but sadly I am very busy with work today so I cannot guarantee that I can fix this until this evening.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/3970dc533574340f58270c6685cbb38caab8d0e1.

maralorn committed:
haskellPackages.haskell-language-server: Fix more deps to stay compat with 1.0.0

sternenseemann committed:
haskellPackages.lzma-static: unbreak

maralorn committed:
haskellPackages.reflex*: Fix various dependencies

peti committed:
hackage2nix: update list of broken packages to fix evaluation errors

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/604db45cba2a8dcd9a7b3ac991e1bfef8c2eb116.

peti commented:
We will merge https://github.com/NixOS/nixpkgs/pull/119652 next week.
